main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,505 @@
+loaded into a vat. The bundling tool uses several standard functions to locate other modules that must 
+be included. These are not a part of SES, but are allowed in module source code, and are translated or 
+removed before execution. 
+
+- `import` and `export` syntax are allowed in ESM-style modules ( preferred over CommonJS). These are 
+  not globals per se, but top-level syntax that defines the module graph.
+- `require`, `module`, `module.exports`, and `exports` are allowed in CommonJS-style modules, and 
+  should work as expected. However, new code should be written as ESM modules. They are either 
+  consumed by the bundling process, provided (in some form) by the execution environment, or 
+  otherwise rewritten to work sensibly
+- `__dirname` and `__filename` are not provided
+- The dynamic import expression (`await import('name')`) is currently prohibited in vat code, 
+  but a future SES implementation may allow it.
+
+Node.js has a [large collection](https://nodejs.org/dist/latest-v14.x/docs/api/) of "built-in 
+modules", such as `http` and `crypto`. Some are clearly platform-specific (e.g. `v8`), while 
+others are not so obvious (`stream`). All are accessed by importing a module (`const v8 = require('v8')` 
+in CommonJS modules, or `import v8 from 'v8'` in ESM modules). These modules are built out of 
+native code (C++), not plain JavaScript.
+
+None of these built-in modules are available to SES code. `require` or `import` can be used on 
+pure JavaScript modules, but not on modules including native code. To exercise authority from 
+a built-in module, you have to write a *device* with an endowment with the built-in module's 
+functions, then send messages to the device.
+
+Browser environments also have a huge list of [other features](https://developer.mozilla.org/en-US/docs/Web/API) 
+presented as names in the global scope (some also added to Node.js). None are available in a 
+SES environment. The most surprising removals include `atob`, `TextEncoder`, and `URL`.
+
+`debugger` is a first-class JavaScript statement, and behaves as expected.
+
+## What does SES add to standard Javascript
+
+- `console` is available to help with debugging. Since all JavaScript implementations
+  add it, you may be surprised it’s not in the official spec. So leaving it out would cause 
+  too much confusion. Note that `console log`’s exact behavior is up to the host program; display 
+  to the operator is not guaranteed. Also, it is difficult at best to write to `process.stdout` and
+  so that should not be done.
+
+- `lockdown()` and `harden()` both an object’s API surface (enumerable data properties). 
+  A hardened object’s properties cannot be changed, so the only way to interact with a 
+  hardened object is through its methods. `harden()` is similar to `Object.freeze()` but more 
+  powerful. See the individual [`lockdown()`](#lockdown) and [`harden()`](#harden) sections
+  below. 
+
+- `HandledPromise` is also a global. 
```

Comment:
Is `HandledPromise` part of SES? or tildot? Not by my understanding.

Are we trying to document SES or the Agoric vat code runtime environment here?

Reply:
Yep. This is something I was aware of with respect to the review. Some of this is from a paper of Brian's that I edited, and I'm not sure which parts were SES specific and which a different part of Agoric's development environment/platform. I figured y'all would be able to tell me which bits don't belong in the SES docs. : -)

Reply:
Concur: `HandledPromise` is specific to the Agoric programming environment, and will also be part of the Endo programming environment, and is not SES.